### PR TITLE
Spring category

### PR DESCRIPTION
**Pull request by keda-vlad**

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/controller/CategoryController.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/category")
+@Tag(name = "Category management", description = "Endpoint for managing categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @PostMapping
+    //@PreAuthorize("hasRole('ROLE_ADMIN')")
```
Comment (DmytroV95) [inline src/main/java/mate/academy/bookstore/controller/CategoryController.java]: You need to uncomment it to ensure admin-only access to create a category.

File: src/main/resources/db/changelog/changes/07-create-categories-table.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+                  name: name
+                  type: varchar(255)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (DmytroV95) [inline src/main/resources/db/changelog/changes/07-create-categories-table.yaml]: No empty line.

File: src/main/resources/db/changelog/changes/08-create-book-category-table.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+                  type: bigint
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: category_id
+                  type: bigint
+                  constraints:
+                    primaryKey: true
+                    nullable: false
```
Comment (DmytroV95) [inline src/main/resources/db/changelog/changes/08-create-book-category-table.yaml]: No empty line.

File: src/main/resources/db/changelog/changes/09-add-description-to-category.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  - changeSet:
+      id: add-description-to-category
+      author: kvlad
+      changes:
+        - addColumn:
+            tableName: categories
+            columns:
+              - column:
+                  name: description
+                  type: varchar(255)
```
Comment (DmytroV95) [inline src/main/resources/db/changelog/changes/09-add-description-to-category.yaml]: No empty line.

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 5]
```diff
@@ -11,3 +11,9 @@ databaseChangeLog:
       file: db/changelog/changes/05-add-roles.yaml
   - include:
       file: db/changelog/changes/06-add-admin.yaml
+  - include:
+      file: db/changelog/changes/07-create-categories-table.yaml
+  - include:
+      file: db/changelog/changes/08-create-book-category-table.yaml
+  - include:
+      file: db/changelog/changes/09-add-description-to-category.yaml
```
Comment (DmytroV95) [inline src/main/resources/db/changelog/db.changelog-master.yaml]: No empty line.

**Review by DmytroV95 — CHANGES_REQUESTED** [review]
Good job! Left some comments to improve your solution.

**Review by DmytroV95 — APPROVED** [review]
LGTM. GJ!

File: src/main/resources/db/changelog/changes/08-create-book-category-table.yaml [hunk 6]
```diff
@@ -0,0 +1,20 @@
+databaseChangeLog:
+  - changeSet:
+      id: create-book-category-table
+      author: kvlad
+      changes:
```
Comment (vovasalyha) [inline src/main/resources/db/changelog/changes/08-create-book-category-table.yaml:5]: add foreign key constraints

**Review by vovasalyha — APPROVED** [review]
Approving, but add foreign key constraints

**Pull request merged**